idea 2023.1, Kotlin 1.8.0, Gradle 8.1.1

JT122406 opened the pull request.

JT122406 committed:
Gradle 8.1

JT122406 committed:
Intellij 2023.1 and Kotlin 1.8.0 update

JT122406 committed:
Slight change to isNullorEmpty()

Juuxel commented:
The build is failing and there are verification issues:

> \> Task :verifyPluginConfiguration
> [gradle-intellij-plugin :verifyPluginConfiguration] The following plugin configuration issues were found:
> - The 'since-build' property is lower than the target IntelliJ Platform major version: 221 < 231.
> - The Kotlin configuration specifies apiVersion=1.8 but since-build='221' property requires apiVersion=1.6.20.
> - The Java configuration specifies sourceCompatibility=11 but IntelliJ Platform 2023.1 requires sourceCompatibility=17.
> - The dependency on the Kotlin Standard Library (stdlib) is automatically added when using the Gradle Kotlin plugin and may > conflict with the version provided with the IntelliJ Platform, see: https://jb.gg/intellij-platform-kotlin-stdlib
> See: https://jb.gg/intellij-platform-versions

JT122406 committed:
Fix build and warnings, add toolchain

JT122406 commented:
Build is fixed, can't get rid of this warning
The Kotlin configuration specifies apiVersion=1.8 but since-build='231' property requires apiVersion=1.8.0
even when I specify 1.8.0 hard to the properties I get a 
![Screenshot (73)](https://user-images.githubusercontent.com/75751882/232327885-2d6a74c6-c50f-4c23-b7e5-04670e38f61e.png)

Which makes no sense but you know Gradle

Juuxel commented on build.gradle.kts:
```diff
@@ -52,3 +52,5 @@ kotlinter {
     )
 }
+
+java.toolchain.languageVersion.set(JavaLanguageVersion.of(17))
```
We don't use the toolchain feature in any other project either, the build should just use the user JVM.

JT122406 replied:
In this case with kotlin it makes sense to have it because of situations like this:
On my system I have Java 17, 19, 20 on the system where it defaults to another version you get an error saying 

Execution failed for task ':compileKotlin'.
> 'compileJava' task (current target is 20) and 'compileKotlin' task (current target is 17) jvm target compatibility should be set to the same Java version.
  Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain

So in cases like this it just makes sense to have this but if you want it removed I can remove it

Juuxel replied:
You can just set the java release
```kotlin
tasks {
  withType<JavaCompile> {
    options.release.set(17)
  }
}
```

Juuxel commented on gradle.properties:
```diff
@@ -1 +1,2 @@
+org.gradle.jvmargs=-Xmx4G
```
Did you test with lower amounts of memory? 1G is enough for Minecraft mods, for example, or 2G for Architectury.

JT122406 committed:
decrease memory to 512M

JT122406 committed:
1G

For some reason 512M built on my PC but not on the workflow but you know funny Github

JT122406 committed:
Remove ToolChain

JT122406 committed:
Gradle 8.1.1

Signed-off-by: Joseph T. McQuigg <J.T.McQuigg12@gmail.com>
